plugin/core/sessions.py

```diff
@@ -85,7 +85,7 @@ def get_initialize_params(workspace_folders: List[WorkspaceFolder], designated_f
             }
         }
     }
-    if config.init_options:
+    if config.init_options is not None:
```

Comment:
`ClientConfig` defaults to `dict` though, right? So this will in practice never be None.

I think the most spec-compliant way would be to not default to `dict` in `ClientSettings` since per-spec the value can be `any`thing.

But this way is more compatible probably, given the wild west out there.

Reply:
You're right, I hadn't thought of ClientConfig having a default value of `{}` for `init_options`.

So the behavior previously was that we don't send an initializationOptions if the dict was empty.
The resulting behavior due to this diff now is that we *always* send initializationOptions, except when the user puts `"initializationOptions": null` in LSP.sublime-settings. If the user leaves out that key LSP will send `"initializationOptions": {}`.

It would be a little neater if init_options were None by default. But like you mentioned, changes to these parts of the code are very sensitive, so let's do that in another PR.